feat: containerd configuration for mirror registry (#292)
d2iq-labs/capi-runtime-extensions: docs/content/customization/generic/image-registries.md

      - name: clusterConfig
        value:
          imageRegistries:
            - url: https://my-registry.io
              credentials:
                secretRef:
                  name: my-registry-credentials
```

Applying this configuration will result in new files and preKubeadmCommands
on the `KubeadmControlPlaneTemplate` and `KubeadmConfigTemplate`.

> To use a image registry as mirror with CA certificate, specify the following configuration:

Reviewer: Great docs!

Author: Thank you

> If your registry mirror requires self signed CA certifate, create a Kubernetes Secret with keys for `ca.crt`:

Reviewer: Maybe something like this since its just 1 key?
```suggestion
If your registry mirror requires self signed CA certifate, create a Kubernetes Secret with its contents and key `ca.crt`:
```
